### PR TITLE
Added Fission References Page , re-arranged pages & menu

### PR DESCRIPTION
**Pull request by techmaharaj**
Added a new page for Fission References with the following changes:

- moved existing documents - Fission CLI Reference ,Fission CRD Reference under it
- modified the url structure for the above pages so that everything comes under `docs -> References` page
- added Fission Glossary page under Reference
- modified the `sidebar_menu_compact` to `true` to show a compact menu. Reason: Fission CLI reference and Releases page have lots of child pages which show up in the sidebar menu. When `sidebar_menu_compact` is set to `true`, it will show the compact menu by default and will expand only when a particular item is clicked.
- changed the weights for main pages so that they appear in order in the navigation menu

### CODE REVIEW

**Comment by sanketsudake** [comment]
Related changes in fission repo https://github.com/fission/fission/pull/2339

File: content/en/docs/reference/glossary.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+Specifies how frequently user traffic needs to be incremented for the new version of function
+
+## E
+
+---
+
+### Environment
+
+Environments provide a run time which is used to execute Fission functions. Currently Fission supports pre-built environments that include NodeJS, Python3, Go, JVM, Ruby, Binary, PHP, .NET and Pearl.
```
[the remark this answers is not in the document]
Comment (sanketsudake) [reply content/en/docs/reference/glossary.md:90]: please fix

File: content/en/docs/reference/glossary.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+---
+
+### Failure Threshold - *Canary Config*
+
+Specifies the threshold in percentage beyond which the new version of a function is declared unhealthy.
+
+### Failure Type - *Canary Config*
+
+Specifies the parameter for checking the health of the new version of a function. *For now, the only supported type is status-code which is the http status code. So if a function returns a status code other than 200, its considered to be unhealthy.*
```
[the remark this answers is not in the document]
Comment (sanketsudake) [reply content/en/docs/reference/glossary.md:106]: please fix

File: content/en/docs/reference/glossary.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+Source archive is an archive file that contains the source code.
+
+### Specification
+
+Specifications (*specs*) are YAML files that are used to instruct Fission CLI about what objects to create or update.
+
+### Storage Service
+
+The storage service is the home for all archives of packages with sizes larger than 256KB.
```
[the remark this answers is not in the document]
Comment (sanketsudake) [reply content/en/docs/reference/glossary.md:284]: please fix

File: content/en/docs/reference/glossary.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+---
+
+## H
+
+---
+
+### HTTP Trigger / Route
+
+It is a type of trigger that invokes a Fission function whenever there is an HTTP request. You can specify a relative URL and HTTP method for a HTTP trigger.
```
[the remark this answers is not in the document]
Comment (sanketsudake) [reply content/en/docs/reference/glossary.md:140]: please check

**Review by sanketsudake — CHANGES_REQUESTED** [review]

File: content/en/docs/reference/glossary.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (techmaharaj) [reply content/en/docs/reference/glossary.md]: fixed

File: content/en/docs/reference/glossary.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (techmaharaj) [reply content/en/docs/reference/glossary.md]: fixed

File: content/en/docs/reference/glossary.md [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+
+RequestsPerPod denotes how many requests will be routed to each pod in Pool Manager.
+
+### Requirements.txt
+
+This is a file that lists down all the external 3rd party modules, libraries that are required for your function to execute.
+
+### Router
+
+Router is responsible for forwarding HTTP requests to functions pods.
```
[the remark this answers is not in the document]
Comment (techmaharaj) [reply content/en/docs/reference/glossary.md]: fixed

File: content/en/docs/reference/glossary.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (techmaharaj) [reply content/en/docs/reference/glossary.md]: fixed

File: content/en/docs/reference/glossary.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (techmaharaj) [reply content/en/docs/reference/glossary.md]: fixed

**Review by techmaharaj — COMMENTED** [review]
Fixed all typos and punctuation.

**Pull request merged**